src/java/main/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -1001,25 +1010,14 @@ public String toString() {
     @Override
     public void close() {
         synchronized(factory.cnxns){
```

Comment:
@CheneySun Please let me know if what you think regarding my comment about removing the excessive synchronization here.